### ISSUE TITLE
Exception when parsing mobile page with spinwheel in hidden view

### ISSUE THREAD

**Issue by billreed63** [maqetta/maqetta#3584]
I get exception when parsing this page...

```
dojo.require('dojox.mobile.deviceTheme');
dojo.require('dojox.mobile.compat');
dojo.require('dojo.data.ItemFileReadStore');
dojo.require('dojox.mobile.RoundRectDataList');
dojo.require('dojox.mobile.SpinWheel');
dojo.require('dojox.mobile.SpinWheelDatePicker');</script>
<script type="text/javascript" src="maqetta/States.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no"></meta>
<script type="text/javascript">
require([
  "maqetta/space",
  "maqetta/AppStates",
  "dojox/io/xhrScriptPlugin",
  "dojo/data/ItemFileReadStore",
  "dojox/mobile",
  "dojox/mobile/deviceTheme",
  "dojox/mobile/compat",
  "dojo/hash",
  "dojox/mobile/RoundRectDataList",
  "dojox/mobile/SpinWheel",
  "dojox/mobile/SpinWheelDatePicker"
]);
</script>
<script type="text/javascript" src="app.js"></script>
</head>
<body class="claro" dvFlowLayout="true" data-davinci-ws="collapse" id="myapp" data-maqetta-device="android_340x480">
 <span dojoType="dojo.data.ItemFileReadStore" id="ItemFileReadStore_1" jsId="ItemFileReadStore_1" data="{&quot;identifier&quot;:&quot;label&quot;,&quot;items&quot;:[{&quot;label&quot;:&quot;View 1&quot;,&quot;moveTo&quot;:&quot;View_1&quot;},{&quot;label&quot;:&quot;Label 2&quot;,&quot;moveTo&quot;:&quot;transitionToPage&quot;}]}"></span><div id="View_0" dojoType="dojox.mobile.View" style="height: 100%;" selected="true">
  <h1 label="Home" dojoType="dojox.mobile.Heading"></h1>
  <ul dojoType="dojox.mobile.RoundRectDataList" store="ItemFileReadStore_1" query="{&quot;label&quot;:&quot;*&quot;}"></ul>
  <ul dojoType="dojox.mobile.RoundRectList">
    <li label="Item 1" dojoType="dojox.mobile.ListItem" moveTo="View_1" sync="true"></li>
    <li label="Item 2" dojoType="dojox.mobile.ListItem" sync="true"></li>
  </ul>
</div>
<div id="View_1" dojoType="dojox.mobile.View" style="display: none; height: 100%;">
<h1 label="View 1" dojoType="dojox.mobile.Heading" back="Home" moveTo="View_0"></h1>
<div dojoType="dojox.mobile.SpinWheel">
<div dojoType="dojox.mobile.SpinWheelSlot" labels="[&quot;A&quot;,&quot;B&quot;,&quot;C&quot;,&quot;D&quot;,&quot;E&quot;]"></div>
<div dojoType="dojox.mobile.SpinWheelSlot" style="height: 453px;" labelFrom="1" labelTo="99"></div>
<div dojoType="dojox.mobile.SpinWheelSlot" labels="[&quot;Dogs&quot;,&quot; cats&quot;,&quot; frogs&quot;]" style="height: 453px;"></div>
</div>
<div dojoType="dojox.mobile.SpinWheelDatePicker"></div>
</div>
</body>
</html>
```

**Comment by billreed63** [comment]
Note this was on a test machine with what looks like a built dojo

**Comment by jhpedemonte** [comment]
What's the exception?  Client or server side?

**Comment by billreed63** [comment]
```
dojo/parser::parse() error

TypeError: _4 is null
[Break On This Error]   

var _5=_4.getMonth();

date.js (line 11)

parser.js (line 411)

_3.getDaysInMonth@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/date.js:11 .onDaySet@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojox/mobile/_DatePickerMixin.js:73 .reset@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojox/mobile/_DatePickerMixin.js:56 .startup@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojox/mobile/_PickerBase.js:17 _2cf@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:3023 .startup@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojox/mobile/SpinWheel.js:11 .cache.dijit/_WidgetBase/</<.startup/<@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dijit/dijit.js:546 .cache.dojo/_base/array/</_578.forEach@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:6781 .cache.dijit/_WidgetBase/</<.startup@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dijit/dijit.js:544 _2cf@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:3023 .cache.dojox/mobile/View/</<.startup@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojox/mobile.js:823 _1b._instantiate/<@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/parser.js:79 .cache.dojo/_base/array/</_578.forEach@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:6781 _1b._instantiate@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/parser.js:77 _1b.parse/p<@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/parser.js:409 .cache.dojo/Deferred/</_4b8@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:5850 .cache.dojo/Deferred/</_4b7/_4c5.then@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:5966 _1b.parse@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/parser.js:408 .cache.davinci/ve/Context/</<._processWidgets/</<@https://maqetta-test.austin.ibm.com:50000/maqetta/app/davinci/davinci.js?bd068a35c05e39f0b2511b65b390c27f33aac7da:57121 .cache.dojo/ready/</dojo.addOnLoad/_5de<@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7273 .cache.dojo/ready/</_5da@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7246 .cache.dojo/ready/</_5d6@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7260 .cache.dojo/ready/</_5da@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7253 .cache.dojo/ready/</_5d6@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7260 .cache.dojo/ready/</_5da@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7253 .cache.dojo/ready/</_5d6@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7260 .cache.dojo/ready/</_5da@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7253 .cache.dojo/ready/</_5d6@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7260 .cache.dojo/ready/</_5da@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7253 .cache.dojo/ready/</_5d6@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7260 .cache.dojo/ready/</_5da@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7253 .cache.dojo/ready/</_5d6@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7260 .cache.dojo/ready/</_5da@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7253 .cache.dojo/ready/</_5d6@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7260 .cache.dojo/ready/</_5da@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7253 .cache.dojo/ready/</_5d6@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7260 .cache.dojo/ready/</_5da@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7253 .cache.dojo/ready/</_5d6@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7260 .cache.dojo/ready/</_5da@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7253 .cache.dojo/ready/</_5d6@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7260 .cache.dojo/ready/</_5da@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7253 .cache.dojo/ready/</_5d6@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7260 .cache.dojo/ready/</_5da@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:7253 req.signal/<@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:153 _9@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:31 req.signal@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:152 _37@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:706 _7b@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:712 _32/_ee@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:822 req.injectUrl/_108@https://maqetta-test.austin.ibm.com:50000/maqetta/user/C/ws/workspace/project1/lib/dojo/dojo/dojo.js:981 

```

**Comment by billreed63** [comment]
This seems to be caused by `<div data-dojo-type="dojox.mobile.SpinWheelDatePicker"></div>` being defined in a mobile view that is hidden at page load. this seems to be a problem with maqetta and not dojo as the page seems to load correctly in preview.

This seems to be a regression, as the code works on 
Version: 7
Date: Nov 11, 2012 8:45:31 PM
Build: c7a79624a47c023...

**Comment by jhpedemonte** [comment]
@JonFerraiolo @peller, either of you guys know what could have caused this regression?

**Comment by peller** [comment]
Can we test to see if this regressed between Dojo 1.7 and 1.8, and if so report it on bugs.dojotoolkit.org?

**Comment by peller** [comment]
does it make any difference if those dojo.requires are removed?

**Comment by peller** [comment]
is this related? #3423

**Comment by billreed63** [comment]
The html is not well formed, I somehow managed to delete the first view so the 

When I have the correct html it works..

**Comment by billreed63** [comment]
new mobile app, drop a View then SpinWheelDatePicker.  save, close, reopen.  get dojo.parse error.
This only happens on the built version of code, running from source is ok...

**Comment by peller** [comment]
SpinWheelSlot's getters call getCenterItem which depends on layout.  This creates a race condition between the startup sequence (which should not have any layout references) and page loading, probably exacerbated by our use of deviceTheme.js

**Comment by peller** [comment]
opened a [Dojo ticket](http://bugs.dojotoolkit.org/ticket/16652) to track this issue

**Comment by peller** [comment]
experiment with a better patch in 93e94c8

**Comment by peller** [comment]
patched in M9, better patch in M10, will be fixed in Dojo 1.9+, possibly 1.8.4.  We can take out the workaround in 93e94c8 if we're satisfied that all users have migrated to a newer Dojo release.
